PWGHF: Implementation of Event mixing for D+-hadron Correlation

Simple-Shyam opened the pull request:
I implemented the event mixing for D+-hadron correlation in Data and MCRec mode.

vkucera commented:
Please fix the formatting by merging the automatic PR (as explained in the [contribution documentation](https://aliceo2group.github.io/analysis-framework/docs/gettingstarted/contributingtocode.html).

Simple-Shyam commented:
> merging the automatic PR (as

Thank you for pointing out I was doing it manually by look at one by one. It will be very helpful.

vkucera commented on PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:
```diff
@@ -54,7 +58,86 @@ const int ptDAxisBins = 180;
+
+  void processDplusSelectionData(aod::Collision const& collision, soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi> const& candidates)
+  {
+    int isDplusFound = 0;
```
Would be better to change to `bool`.

Simple-Shyam replied:
It's already converted to bool.

vkucera commented on PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:
```diff
@@ -54,7 +58,86 @@ const int ptDAxisBins = 180;
+        if (ptCandMin >= 0. && candidate1.pt() < ptCandMin) {
+          continue;
+        }
+        isDplusFound = 1;
+      }
```
You can already exit the loop at this point.

Simple-Shyam replied:
break is already done in the loop.

vkucera commented on PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:
```diff
@@ -54,7 +58,86 @@ const int ptDAxisBins = 180;
+        if (ptCandMin >= 0. && candidate1.pt() < ptCandMin) {
+          continue;
+        }
+        isDplusFound = 1;
+      }
```
Same as above

Simple-Shyam replied:
Already done.

vkucera commented on PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:
```diff
@@ -54,7 +58,86 @@ const int ptDAxisBins = 180;
+    if (selectedDPlusCandidates.size() > 0) {
+      auto selectedDPlusCandidatesGrouped = selectedDPlusCandidates->sliceByCached(aod::hf_cand::collisionId, collision.globalIndex());
+
+      for (auto& candidate1 : selectedDPlusCandidatesGrouped) {
```
```suggestion
      for (auto const& candidate1 : selectedDPlusCandidatesGrouped) {
```

Simple-Shyam replied:
I used the const keyword as suggested.

vkucera commented on PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:
```diff
@@ -54,7 +58,86 @@ const int ptDAxisBins = 180;
+    int isDplusFound = 0;
+    if (recoFlagDPlusCandidates.size() > 0) {
+      auto selectedDPlusCandidatesGroupedMC = recoFlagDPlusCandidates->sliceByCached(aod::hf_cand::collisionId, collision.globalIndex());
+      for (auto& candidate1 : selectedDPlusCandidatesGroupedMC) {
```
```suggestion
      for (auto const& candidate1 : selectedDPlusCandidatesGroupedMC) {
```

Simple-Shyam replied:
Already applied.

vkucera commented on PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:
```diff
@@ -54,7 +58,86 @@ const int ptDAxisBins = 180;
+  void processDplusSelectionMCGen(aod::McCollision const& mccollision, aod::McParticles const& particlesMC)
+  {
+    int isDplusFound = 0;
+    for (auto& particle1 : particlesMC) {
```
```suggestion
    for (auto const& particle1 : particlesMC) {
```

Simple-Shyam replied:
Yes already applied.

vkucera commented on PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:
```diff
@@ -54,7 +58,86 @@ const int ptDAxisBins = 180;
+      if (ptCandMin >= 0. && particle1.pt() < ptCandMin) {
+        continue;
+      }
+      isDplusFound = 1;
```
Same as above

vkucera commented on PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:
```diff
@@ -54,7 +58,86 @@ const int ptDAxisBins = 180;
+  }
+  PROCESS_SWITCH(HfDplusSelection, processDplusSelectionMCRec, "Process Dplus Selection MCRec", false);
+
+  void processDplusSelectionMCGen(aod::McCollision const& mccollision, aod::McParticles const& particlesMC)
```
```suggestion
  void processDplusSelectionMcGen(aod::McCollision const& mccollision, aod::McParticles const& particlesMc)
```

Simple-Shyam replied:
Yes fixed.

vkucera commented on PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:
```diff
@@ -54,7 +58,86 @@ const int ptDAxisBins = 180;
+  {
+    int isDplusFound = 0;
+    if (recoFlagDPlusCandidates.size() > 0) {
+      auto selectedDPlusCandidatesGroupedMC = recoFlagDPlusCandidates->sliceByCached(aod::hf_cand::collisionId, collision.globalIndex());
```
```suggestion
      auto selectedDplusCandidatesGroupedMc = recoFlagDPlusCandidates->sliceByCached(aod::hf_cand::collisionId, collision.globalIndex());
```

Simple-Shyam replied:
Yes fixed.

vkucera commented on PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:
```diff
@@ -54,7 +58,86 @@ const int ptDAxisBins = 180;
+  }
+  PROCESS_SWITCH(HfDplusSelection, processDplusSelectionData, "Process Dplus Selection Data", false);
+
+  void processDplusSelectionMCRec(aod::Collision const& collision, soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi, aod::HfCand3ProngMcRec> const& candidates)
```
```suggestion
  void processDplusSelectionMcRec(aod::Collision const& collision, soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi, aod::HfCand3ProngMcRec> const& candidates)
```

Simple-Shyam replied:
Done.

vkucera commented on PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:
```diff
@@ -54,7 +58,86 @@ const int ptDAxisBins = 180;
+  Configurable<double> yCandMax{"yCandMax", 0.8, "max. cand. rapidity"};
+  Configurable<double> ptCandMin{"ptCandMin", 1., "min. cand. pT"};
+
+  Partition<soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi>> selectedDPlusCandidates = aod::hf_sel_candidate_dplus::isSelDplusToPiKPi >= selectionFlagDplus;
+  Partition<soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi, aod::HfCand3ProngMcRec>> recoFlagDPlusCandidates = aod::hf_sel_candidate_dplus::isSelDplusToPiKPi > 0;
```
```suggestion
  Partition<soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi>> selectedDplusCandidates = aod::hf_sel_candidate_dplus::isSelDplusToPiKPi >= selectionFlagDplus;
  Partition<soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi, aod::HfCand3ProngMcRec>> recoFlagDplusCandidates = aod::hf_sel_candidate_dplus::isSelDplusToPiKPi > 0;
```

Simple-Shyam replied:
Done

vkucera commented on PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:
```diff
@@ -54,7 +58,86 @@ const int ptDAxisBins = 180;
+  Configurable<double> yCandMax{"yCandMax", 0.8, "max. cand. rapidity"};
+  Configurable<double> ptCandMin{"ptCandMin", 1., "min. cand. pT"};
+
+  Partition<soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi>> selectedDPlusCandidates = aod::hf_sel_candidate_dplus::isSelDplusToPiKPi >= selectionFlagDplus;
+  Partition<soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi, aod::HfCand3ProngMcRec>> recoFlagDPlusCandidates = aod::hf_sel_candidate_dplus::isSelDplusToPiKPi > 0;
```
Why do you apply different selection in real data and MC rec. data?

Simple-Shyam replied:
I used the selectionFlagDplus in both cases now.

vkucera commented on PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:
```diff
@@ -282,20 +375,37 @@ struct HfCorrelatorDplusHadrons {
+
+    auto getTracksSize = [&particlesMC](aod::McCollision const& collision) {
+      int nTracks = 0;
+      for (auto& track : particlesMC) {
```
```suggestion
      for (auto const& track : particlesMC) {
```

Simple-Shyam replied:
done

vkucera commented on PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:
```diff
@@ -282,20 +375,37 @@ struct HfCorrelatorDplusHadrons {
   /// Dplus-Hadron correlation pair builder - for MC gen-level analysis (no filter/selection, only true signal)
-  void processMcGen(aod::McCollision const& mccollision, MCParticlesPlus3Prong const& particlesMC)
+
+  void processMcGen(aod::McCollision const& mccollision, aod::McParticles const& particlesMC)
```
```suggestion
  void processMcGen(aod::McCollision const& mccollision, aod::McParticles const& particlesMc)
```

Simple-Shyam replied:
mccollision->mcCollision

vkucera commented on PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:
```diff
@@ -54,7 +58,86 @@ const int ptDAxisBins = 180;
+struct HfDplusSelection {
+  Produces<aod::DmesonSelection> dplusSel;
+  Configurable<int> selectionFlagDplus{"selectionFlagDplus", 1, "Selection Flag for Dplus"};
+  Configurable<double> yCandMax{"yCandMax", 0.8, "max. cand. rapidity"};
+  Configurable<double> ptCandMin{"ptCandMin", 1., "min. cand. pT"};
```
`float` should be satisfactory.

Simple-Shyam replied:
done

vkucera replied:
They are still `double`.

vkucera commented on PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:
```diff
@@ -331,19 +453,81 @@ struct HfCorrelatorDplusHadrons {
+  Filter collisionFilter = aod::hf_selection_dmeson_collision::dmesonsel > 0;
+  using mySelCollisions = soa::Filtered<myCollisions>;
+
+  Filter trackFilter = (aod::track::eta > static_cast<float>(-etaTrackMax)) && (aod::track::eta < static_cast<float>(etaTrackMax)) && (aod::track::pt > static_cast<float>(ptTrackMin)) && (aod::track::dcaXY > static_cast<float>(-dcaXYTrackMax)) && (aod::track::dcaXY < static_cast<float>(dcaXYTrackMax)) &&
+                       (aod::track::dcaZ > static_cast<float>(-dcaZTrackMax)) && (aod::track::dcaZ < static_cast<float>(dcaZTrackMax));
```
Make the configurables `float` so that you don't have to cast them and use `nabs` for the absolute value filters.

Simple-Shyam replied:
done

vkucera commented on PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:
```diff
@@ -331,19 +453,81 @@ struct HfCorrelatorDplusHadrons {
   PROCESS_SWITCH(HfCorrelatorDplusHadrons, processMcGen, "Process MC Gen mode", false);
+
+  // Event Mixing for the Data Mode
+  using myCollisions = soa::Join<aod::Collisions, aod::Mults, aod::DmesonSelection>;
+  Filter collisionFilter = aod::hf_selection_dmeson_collision::dmesonsel > 0;
+  using mySelCollisions = soa::Filtered<myCollisions>;
+
+  Filter trackFilter = (aod::track::eta > static_cast<float>(-etaTrackMax)) && (aod::track::eta < static_cast<float>(etaTrackMax)) && (aod::track::pt > static_cast<float>(ptTrackMin)) && (aod::track::dcaXY > static_cast<float>(-dcaXYTrackMax)) && (aod::track::dcaXY < static_cast<float>(dcaXYTrackMax)) &&
+                       (aod::track::dcaZ > static_cast<float>(-dcaZTrackMax)) && (aod::track::dcaZ < static_cast<float>(dcaZTrackMax));
+
+  using fullTracks = soa::Join<aod::Tracks, aod::TracksDCA>;
+  using myTracks = soa::Filtered<fullTracks>;
+
+  Filter dplusfilter = aod::hf_sel_candidate_dplus::isSelDplusToPiKPi >= 1;
+  using myCandidatesData = soa::Filtered<soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi>>;
```
Fix the order of struct members.

Simple-Shyam replied:
What does this mean @vkucera? I didn't understand it clearly.  I already fixed the other comments.

vkucera replied:
The members should be grouped and sorted. Please see the HF guidelines.

Simple-Shyam replied:
done following the guideline given in documentation.

vkucera reviewed (changes requested):
There is `const` missing in many loops.

Simple-Shyam commented on PWGHF/DataModel/CandidateReconstructionTables.h:
```diff
@@ -1290,15 +1290,24 @@ DECLARE_SOA_COLUMN(PtD, ptD, float);
+// Table for selection of Dmeson in a collision
+namespace hf_selection_dmeson_collision
+{
+DECLARE_SOA_COLUMN(DmesonSel, dmesonsel, int);
```
> issing in many loops.

Thanks Vit,
I will fix these comments in the coming days. There is a general comment that there was some compilation issue if we put const in the process loop in the event mixing @saganatt.  Once it's fixed then I can put const otherwise the code will not compile for the mixed events.

alibuild commented:
Error while checking build/O2Physics/o2 for 791bef733ef0558b52dda12f3c70c75d8f56cfcb at 2023-04-19 16:22:
```
## sw/BUILD/O2Physics-latest/log
/sw/SOURCES/O2Physics/2329-slc7_x86-64/0/PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:82:83: error: no matching function for call to 'o2::soa::Filtered<o2::soa::Join<o2::soa::Join<o2::aod::HfCand3ProngExtExtension, o2::soa::Table<o2::soa::Index<0, -1>, o2::aod::hf_cand::CollisionId, o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex, o2::aod::hf_cand::ErrorDecayLength, o2::aod::hf_cand::ErrorDecayLengthXY, o2::aod::hf_cand::Chi2PCA, o2::aod::hf_cand::RSecondaryVertex<o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex>, o2::aod::hf_cand::DecayLength<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex>, o2::aod::hf_cand::DecayLengthXY<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex>, o2::aod::hf_cand::DecayLengthNormalised<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex, o2::aod::hf_cand::ErrorDecayLength>, o2::aod::hf_cand::DecayLengthXYNormalised<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ErrorDecayLengthXY>, o2::aod::hf_cand::ImpactParameterNormalised0<o2::aod::hf_cand::ImpactParameter0, o2::aod::hf_cand::ErrorImpactParameter0>, o2::aod::hf_cand::PtProng0<o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0>, o2::aod::hf_cand::Pt2Prong0<o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0>, o2::aod::hf_cand::PVectorProng0<o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0, o2::aod::hf_cand::PzProng0>, o2::aod::hf_cand::ImpactParameterNormalised1<o2::aod::hf_cand::ImpactParameter1, o2::aod::hf_cand::ErrorImpactParameter1>, o2::aod::hf_cand::PtProng1<o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1>, o2::aod::hf_cand::Pt2Prong1<o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1>, o2::aod::hf_cand::PVectorProng1<o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1, o2::aod::hf_cand::PzProng1>, o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0, o2::aod::hf_cand::PzProng0, o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1, o2::aod::hf_cand::PzProng1, o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2, o2::aod::hf_cand::PzProng2, o2::aod::hf_cand::ImpactParameter0, o2::aod::hf_cand::ImpactParameter1, o2::aod::hf_cand::ImpactParameter2, o2::aod::hf_cand::ErrorImpactParameter0, o2::aod::hf_cand::ErrorImpactParameter1, o2::aod::hf_cand::ErrorImpactParameter2, o2::aod::hf_track_index::Prong0Id, o2::aod::hf_track_index::Prong1Id, o2::aod::hf_track_index::Prong2Id, o2::aod::hf_track_index::HFflag, o2::aod::hf_cand_3prong::M<o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0, o2::aod::hf_cand::PzProng0, o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1, o2::aod::hf_cand::PzProng1, o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2, o2::aod::hf_cand::PzProng2>, o2::aod::hf_cand_3prong::M2<o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0, o2::aod::hf_cand::PzProng0, o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1, o2::aod::hf_cand::PzProng1, o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2, o2::aod::hf_cand::PzProng2>, o2::aod::hf_cand_3prong::ImpactParameterProngSqSum<o2::aod::hf_cand::ImpactParameter0, o2::aod::hf_cand::ImpactParameter1, o2::aod::hf_cand::ImpactParameter2>, o2::aod::hf_cand::ImpactParameterNormalised2<o2::aod::hf_cand::ImpactParameter2, o2::aod::hf_cand::ErrorImpactParameter2>, o2::aod::hf_cand::PtProng2<o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2>, o2::aod::hf_cand::Pt2Prong2<o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2>, o2::aod::hf_cand::PVectorProng2<o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2, o2::aod::hf_cand::PzProng2>, o2::aod::hf_cand::Pt<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py>, o2::aod::hf_cand::Pt2<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py>, o2::aod::hf_cand::P<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::P2<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::PVector<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::CPA<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex, o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::CPAXY<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py>, o2::aod::hf_cand::Ct<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex, o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::ImpactParameterXY<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex, o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand_3prong::MaxNormalisedDeltaIP<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ErrorDecayLengthXY, o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand::ImpactParameter0, o2::aod::hf_cand::ErrorImpactParameter0, o2::aod::hf_cand::ImpactParameter1, o2::aod::hf_cand::ErrorImpactParameter1, o2::aod::hf_cand::ImpactParameter2, o2::aod::hf_cand::ErrorImpactParameter2, o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0, o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1, o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2>, o2::aod::hf_cand::Eta<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::Phi<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py>, o2::aod::hf_cand::Y<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::E<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::E2<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz> > >, o2::soa::Table<o2::aod::hf_sel_candidate_dplus::IsSelDplusToPiKPi> > >::sliceByCached(const o2::framework::expressions::BindingNode&, int64_t)'
/sw/SOURCES/O2Physics/2329-slc7_x86-64/0/PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:102:85: error: no matching function for call to 'o2::soa::Filtered<o2::soa::Join<o2::soa::Join<o2::aod::HfCand3ProngExtExtension, o2::soa::Table<o2::soa::Index<0, -1>, o2::aod::hf_cand::CollisionId, o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex, o2::aod::hf_cand::ErrorDecayLength, o2::aod::hf_cand::ErrorDecayLengthXY, o2::aod::hf_cand::Chi2PCA, o2::aod::hf_cand::RSecondaryVertex<o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex>, o2::aod::hf_cand::DecayLength<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex>, o2::aod::hf_cand::DecayLengthXY<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex>, o2::aod::hf_cand::DecayLengthNormalised<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex, o2::aod::hf_cand::ErrorDecayLength>, o2::aod::hf_cand::DecayLengthXYNormalised<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ErrorDecayLengthXY>, o2::aod::hf_cand::ImpactParameterNormalised0<o2::aod::hf_cand::ImpactParameter0, o2::aod::hf_cand::ErrorImpactParameter0>, o2::aod::hf_cand::PtProng0<o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0>, o2::aod::hf_cand::Pt2Prong0<o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0>, o2::aod::hf_cand::PVectorProng0<o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0, o2::aod::hf_cand::PzProng0>, o2::aod::hf_cand::ImpactParameterNormalised1<o2::aod::hf_cand::ImpactParameter1, o2::aod::hf_cand::ErrorImpactParameter1>, o2::aod::hf_cand::PtProng1<o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1>, o2::aod::hf_cand::Pt2Prong1<o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1>, o2::aod::hf_cand::PVectorProng1<o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1, o2::aod::hf_cand::PzProng1>, o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0, o2::aod::hf_cand::PzProng0, o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1, o2::aod::hf_cand::PzProng1, o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2, o2::aod::hf_cand::PzProng2, o2::aod::hf_cand::ImpactParameter0, o2::aod::hf_cand::ImpactParameter1, o2::aod::hf_cand::ImpactParameter2, o2::aod::hf_cand::ErrorImpactParameter0, o2::aod::hf_cand::ErrorImpactParameter1, o2::aod::hf_cand::ErrorImpactParameter2, o2::aod::hf_track_index::Prong0Id, o2::aod::hf_track_index::Prong1Id, o2::aod::hf_track_index::Prong2Id, o2::aod::hf_track_index::HFflag, o2::aod::hf_cand_3prong::M<o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0, o2::aod::hf_cand::PzProng0, o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1, o2::aod::hf_cand::PzProng1, o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2, o2::aod::hf_cand::PzProng2>, o2::aod::hf_cand_3prong::M2<o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0, o2::aod::hf_cand::PzProng0, o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1, o2::aod::hf_cand::PzProng1, o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2, o2::aod::hf_cand::PzProng2>, o2::aod::hf_cand_3prong::ImpactParameterProngSqSum<o2::aod::hf_cand::ImpactParameter0, o2::aod::hf_cand::ImpactParameter1, o2::aod::hf_cand::ImpactParameter2>, o2::aod::hf_cand::ImpactParameterNormalised2<o2::aod::hf_cand::ImpactParameter2, o2::aod::hf_cand::ErrorImpactParameter2>, o2::aod::hf_cand::PtProng2<o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2>, o2::aod::hf_cand::Pt2Prong2<o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2>, o2::aod::hf_cand::PVectorProng2<o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2, o2::aod::hf_cand::PzProng2>, o2::aod::hf_cand::Pt<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py>, o2::aod::hf_cand::Pt2<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py>, o2::aod::hf_cand::P<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::P2<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::PVector<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::CPA<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex, o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::CPAXY<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py>, o2::aod::hf_cand::Ct<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex, o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::ImpactParameterXY<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex, o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand_3prong::MaxNormalisedDeltaIP<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ErrorDecayLengthXY, o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand::ImpactParameter0, o2::aod::hf_cand::ErrorImpactParameter0, o2::aod::hf_cand::ImpactParameter1, o2::aod::hf_cand::ErrorImpactParameter1, o2::aod::hf_cand::ImpactParameter2, o2::aod::hf_cand::ErrorImpactParameter2, o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0, o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1, o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2>, o2::aod::hf_cand::Eta<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::Phi<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py>, o2::aod::hf_cand::Y<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::E<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::E2<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz> > >, o2::soa::Table<o2::aod::hf_sel_candidate_dplus::IsSelDplusToPiKPi>, o2::soa::Table<o2::aod::hf_cand_3prong::FlagMcMatchRec, o2::aod::hf_cand_3prong::OriginMcRec, o2::aod::hf_cand_3prong::IsCandidateSwapped, o2::aod::hf_cand_3prong::FlagMcDecayChanRec> > >::sliceByCached(const o2::framework::expressions::BindingNode&, int64_t)'
/sw/SOURCES/O2Physics/2329-slc7_x86-64/0/PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:490:120: error: no matching function for call to 'o2::framework::GroupedCombinationsGenerator<int, o2::soa::CombinationsBlockStrictlyUpperSameIndexPolicy<o2::framework::ColumnBinningPolicy<o2::aod::collision::PosZ, o2::aod::mult::MultFV0M<o2::aod::mult::MultFV0A, o2::aod::mult::MultFV0C> >, int, o2::soa::Filtered<o2::soa::Join<o2::soa::Table<o2::soa::Index<0, -1>, o2::aod::collision::BCId, o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::collision::CovXX, o2::aod::collision::CovXY, o2::aod::collision::CovYY, o2::aod::collision::CovXZ, o2::aod::collision::CovYZ, o2::aod::collision::CovZZ, o2::aod::collision::Flags, o2::aod::collision::Chi2, o2::aod::collision::NumContrib, o2::aod::collision::CollisionTime, o2::aod::collision::CollisionTimeRes>, o2::soa::Table<o2::aod::mult::MultFV0A, o2::aod::mult::MultFV0C, o2::aod::mult::MultFT0A, o2::aod::mult::MultFT0C, o2::aod::mult::MultFDDA, o2::aod::mult::MultFDDC, o2::aod::mult::MultZNA, o2::aod::mult::MultZNC, o2::aod::mult::MultFV0M<o2::aod::mult::MultFV0A, o2::aod::mult::MultFV0C>, o2::aod::mult::MultFT0M<o2::aod::mult::MultFT0A, o2::aod::mult::MultFT0C>, o2::aod::mult::MultFDDM<o2::aod::mult::MultFDDA, o2::aod::mult::MultFDDC>, o2::aod::mult::MultTracklets, o2::aod::mult::MultTPC, o2::aod::mult::MultNTracksPV, o2::aod::mult::MultNTracksPVeta1>, o2::soa::Table<o2::aod::hf_selection_dmeson_collision::DmesonSel> > >, o2::soa::Filtered<o2::soa::Join<o2::soa::Table<o2::soa::Index<0, -1>, o2::aod::collision::BCId, o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::collision::CovXX, o2::aod::collision::CovXY, o2::aod::collision::CovYY, o2::aod::collision::CovXZ, o2::aod::collision::CovYZ, o2::aod::collision::CovZZ, o2::aod::collision::Flags, o2::aod::collision::Chi2, o2::aod::collision::NumContrib, o2::aod::collision::CollisionTime, o2::aod::collision::CollisionTimeRes>, o2::soa::Table<o2::aod::mult::MultFV0A, o2::aod::mult::MultFV0C, o2::aod::mult::MultFT0A, o2::aod::mult::MultFT0C, o2::aod::mult::MultFDDA, o2::aod::mult::MultFDDC, o2::aod::mult::MultZNA, o2::aod::mult::MultZNC, o2::aod::mult::MultFV0M<o2::aod::mult::MultFV0A, o2::aod::mult::MultFV0C>, o2::aod::mult::MultFT0M<o2::aod::mult::MultFT0A, o2::aod::mult::MultFT0C>, o2::aod::mult::MultFDDM<o2::aod::mult::MultFDDA, o2::aod::mult::MultFDDC>, o2::aod::mult::MultTracklets, o2::aod::mult::MultTPC, o2::aod::mult::MultNTracksPV, o2::aod::mult::MultNTracksPVeta1>, o2::soa::Table<o2::aod::hf_selection_dmeson_collision::DmesonSel> > > >, o2::framework::ColumnBinningPolicy<o2::aod::collision::PosZ, o2::aod::mult::MultFV0M<o2::aod::mult::MultFV0A, o2::aod::mult::MultFV0C> >, o2::soa::Filtered<o2::soa::Join<o2::soa::Table<o2::soa::Index<0, -1>, o2::aod::collision::BCId, o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::collision::CovXX, o2::aod::collision::CovXY, o2::aod::collision::CovYY, o2::aod::collision::CovXZ, o2::aod::collision::CovYZ, o2::aod::collision::CovZZ, o2::aod::collision::Flags, o2::aod::collision::Chi2, o2::aod::collision::NumContrib, o2::aod::collision::CollisionTime, o2::aod::collision::CollisionTimeRes>, o2::soa::Table<o2::aod::mult::MultFV0A, o2::aod::mult::MultFV0C, o2::aod::mult::MultFT0A, o2::aod::mult::MultFT0C, o2::aod::mult::MultFDDA, o2::aod::mult::MultFDDC, o2::aod::mult::MultZNA, o2::aod::mult::MultZNC, o2::aod::mult::MultFV0M<o2::aod::mult::MultFV0A, o2::aod::mult::MultFV0C>, o2::aod::mult::MultFT0M<o2::aod::mult::MultFT0A, o2::aod::mult::MultFT0C>, o2::aod::mult::MultFDDM<o2::aod::mult::MultFDDA, o2::aod::mult::MultFDDC>, o2::aod::mult::MultTracklets, o2::aod::mult::MultTPC, o2::aod::mult::MultNTracksPV, o2::aod::mult::MultNTracksPVeta1>, o2::soa::Table<o2::aod::hf_selection_dmeson_collision::DmesonSel> > >, o2::soa::Filtered<o2::soa::Join<o2::soa::Join<o2::aod::HfCand3ProngExtExtension, o2::soa::Table<o2::soa::Index<0, -1>, o2::aod::hf_cand::CollisionId, o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex, o2::aod::hf_cand::ErrorDecayLength, o2::aod::hf_cand::ErrorDecayLengthXY, o2::aod::hf_cand::Chi2PCA, o2::aod::hf_cand::RSecondaryVertex<o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex>, o2::aod::hf_cand::DecayLength<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex>, o2::aod::hf_cand::DecayLengthXY<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex>, o2::aod::hf_cand::DecayLengthNormalised<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex, o2::aod::hf_cand::ErrorDecayLength>, o2::aod::hf_cand::DecayLengthXYNormalised<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ErrorDecayLengthXY>, o2::aod::hf_cand::ImpactParameterNormalised0<o2::aod::hf_cand::ImpactParameter0, o2::aod::hf_cand::ErrorImpactParameter0>, o2::aod::hf_cand::PtProng0<o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0>, o2::aod::hf_cand::Pt2Prong0<o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0>, o2::aod::hf_cand::PVectorProng0<o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0, o2::aod::hf_cand::PzProng0>, o2::aod::hf_cand::ImpactParameterNormalised1<o2::aod::hf_cand::ImpactParameter1, o2::aod::hf_cand::ErrorImpactParameter1>, o2::aod::hf_cand::PtProng1<o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1>, o2::aod::hf_cand::Pt2Prong1<o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1>, o2::aod::hf_cand::PVectorProng1<o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1, o2::aod::hf_cand::PzProng1>, o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0, o2::aod::hf_cand::PzProng0, o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1, o2::aod::hf_cand::PzProng1, o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2, o2::aod::hf_cand::PzProng2, o2::aod::hf_cand::ImpactParameter0, o2::aod::hf_cand::ImpactParameter1, o2::aod::hf_cand::ImpactParameter2, o2::aod::hf_cand::ErrorImpactParameter0, o2::aod::hf_cand::ErrorImpactParameter1, o2::aod::hf_cand::ErrorImpactParameter2, o2::aod::hf_track_index::Prong0Id, o2::aod::hf_track_index::Prong1Id, o2::aod::hf_track_index::Prong2Id, o2::aod::hf_track_index::HFflag, o2::aod::hf_cand_3prong::M<o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0, o2::aod::hf_cand::PzProng0, o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1, o2::aod::hf_cand::PzProng1, o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2, o2::aod::hf_cand::PzProng2>, o2::aod::hf_cand_3prong::M2<o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0, o2::aod::hf_cand::PzProng0, o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1, o2::aod::hf_cand::PzProng1, o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2, o2::aod::hf_cand::PzProng2>, o2::aod::hf_cand_3prong::ImpactParameterProngSqSum<o2::aod::hf_cand::ImpactParameter0, o2::aod::hf_cand::ImpactParameter1, o2::aod::hf_cand::ImpactParameter2>, o2::aod::hf_cand::ImpactParameterNormalised2<o2::aod::hf_cand::ImpactParameter2, o2::aod::hf_cand::ErrorImpactParameter2>, o2::aod::hf_cand::PtProng2<o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2>, o2::aod::hf_cand::Pt2Prong2<o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2>, o2::aod::hf_cand::PVectorProng2<o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2, o2::aod::hf_cand::PzProng2>, o2::aod::hf_cand::Pt<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py>, o2::aod::hf_cand::Pt2<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py>, o2::aod::hf_cand::P<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::P2<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::PVector<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::CPA<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex, o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::CPAXY<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py>, o2::aod::hf_cand::Ct<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex, o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::ImpactParameterXY<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex, o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand_3prong::MaxNormalisedDeltaIP<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ErrorDecayLengthXY, o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand::ImpactParameter0, o2::aod::hf_cand::ErrorImpactParameter0, o2::aod::hf_cand::ImpactParameter1, o2::aod::hf_cand::ErrorImpactParameter1, o2::aod::hf_cand::ImpactParameter2, o2::aod::hf_cand::ErrorImpactParameter2, o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0, o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1, o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2>, o2::aod::hf_cand::Eta<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::Phi<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py>, o2::aod::hf_cand::Y<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::E<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::E2<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz> > >, o2::soa::Table<o2::aod::hf_sel_candidate_dplus::IsSelDplusToPiKPi> > >, o2::soa::Filtered<o2::soa::Join<o2::soa::Join<o2::aod::TracksExtension, o2::soa::Table<o2::soa::Index<0, -1>, o2::aod::track::CollisionId, o2::aod::track::TrackType, o2::aod::track::X, o2::aod::track::Alpha, o2::aod::track::Y, o2::aod::track::Z, o2::aod::track::Snp, o2::aod::track::Tgl, o2::aod::track::Signed1Pt, o2::aod::track::IsWithinBeamPipe<o2::aod::track::X>, o2::aod::track::Px<o2::aod::track::Signed1Pt, o2::aod::track::Snp, o2::aod::track::Alpha>, o2::aod::track::Py<o2::aod::track::Signed1Pt, o2::aod::track::Snp, o2::aod::track::Alpha>, o2::aod::track::Pz<o2::aod::track::Signed1Pt, o2::aod::track::Tgl>, o2::aod::track::Energy<o2::aod::track::Signed1Pt, o2::aod::track::Tgl>, o2::aod::track::Rapidity<o2::aod::track::Signed1Pt, o2::aod::track::Tgl>, o2::aod::track::Sign<o2::aod::track::Signed1Pt>, o2::soa::Marker<1> > >, o2::soa::Table<o2::aod::track::DcaXY, o2::aod::track::DcaZ> > > >::GroupedCombinationsGenerator(<brace-enclosed initializer list>)'
/sw/SOURCES/O2Physics/2329-slc7_x86-64/0/PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:513:122: error: no matching function for call to 'o2::framework::GroupedCombinationsGenerator<int, o2::soa::CombinationsBlockStrictlyUpperSameIndexPolicy<o2::framework::ColumnBinningPolicy<o2::aod::collision::PosZ, o2::aod::mult::MultFV0M<o2::aod::mult::MultFV0A, o2::aod::mult::MultFV0C> >, int, o2::soa::Filtered<o2::soa::Join<o2::soa::Table<o2::soa::Index<0, -1>, o2::aod::collision::BCId, o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::collision::CovXX, o2::aod::collision::CovXY, o2::aod::collision::CovYY, o2::aod::collision::CovXZ, o2::aod::collision::CovYZ, o2::aod::collision::CovZZ, o2::aod::collision::Flags, o2::aod::collision::Chi2, o2::aod::collision::NumContrib, o2::aod::collision::CollisionTime, o2::aod::collision::CollisionTimeRes>, o2::soa::Table<o2::aod::mult::MultFV0A, o2::aod::mult::MultFV0C, o2::aod::mult::MultFT0A, o2::aod::mult::MultFT0C, o2::aod::mult::MultFDDA, o2::aod::mult::MultFDDC, o2::aod::mult::MultZNA, o2::aod::mult::MultZNC, o2::aod::mult::MultFV0M<o2::aod::mult::MultFV0A, o2::aod::mult::MultFV0C>, o2::aod::mult::MultFT0M<o2::aod::mult::MultFT0A, o2::aod::mult::MultFT0C>, o2::aod::mult::MultFDDM<o2::aod::mult::MultFDDA, o2::aod::mult::MultFDDC>, o2::aod::mult::MultTracklets, o2::aod::mult::MultTPC, o2::aod::mult::MultNTracksPV, o2::aod::mult::MultNTracksPVeta1>, o2::soa::Table<o2::aod::hf_selection_dmeson_collision::DmesonSel> > >, o2::soa::Filtered<o2::soa::Join<o2::soa::Table<o2::soa::Index<0, -1>, o2::aod::collision::BCId, o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::collision::CovXX, o2::aod::collision::CovXY, o2::aod::collision::CovYY, o2::aod::collision::CovXZ, o2::aod::collision::CovYZ, o2::aod::collision::CovZZ, o2::aod::collision::Flags, o2::aod::collision::Chi2, o2::aod::collision::NumContrib, o2::aod::collision::CollisionTime, o2::aod::collision::CollisionTimeRes>, o2::soa::Table<o2::aod::mult::MultFV0A, o2::aod::mult::MultFV0C, o2::aod::mult::MultFT0A, o2::aod::mult::MultFT0C, o2::aod::mult::MultFDDA, o2::aod::mult::MultFDDC, o2::aod::mult::MultZNA, o2::aod::mult::MultZNC, o2::aod::mult::MultFV0M<o2::aod::mult::MultFV0A, o2::aod::mult::MultFV0C>, o2::aod::mult::MultFT0M<o2::aod::mult::MultFT0A, o2::aod::mult::MultFT0C>, o2::aod::mult::MultFDDM<o2::aod::mult::MultFDDA, o2::aod::mult::MultFDDC>, o2::aod::mult::MultTracklets, o2::aod::mult::MultTPC, o2::aod::mult::MultNTracksPV, o2::aod::mult::MultNTracksPVeta1>, o2::soa::Table<o2::aod::hf_selection_dmeson_collision::DmesonSel> > > >, o2::framework::ColumnBinningPolicy<o2::aod::collision::PosZ, o2::aod::mult::MultFV0M<o2::aod::mult::MultFV0A, o2::aod::mult::MultFV0C> >, o2::soa::Filtered<o2::soa::Join<o2::soa::Table<o2::soa::Index<0, -1>, o2::aod::collision::BCId, o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::collision::CovXX, o2::aod::collision::CovXY, o2::aod::collision::CovYY, o2::aod::collision::CovXZ, o2::aod::collision::CovYZ, o2::aod::collision::CovZZ, o2::aod::collision::Flags, o2::aod::collision::Chi2, o2::aod::collision::NumContrib, o2::aod::collision::CollisionTime, o2::aod::collision::CollisionTimeRes>, o2::soa::Table<o2::aod::mult::MultFV0A, o2::aod::mult::MultFV0C, o2::aod::mult::MultFT0A, o2::aod::mult::MultFT0C, o2::aod::mult::MultFDDA, o2::aod::mult::MultFDDC, o2::aod::mult::MultZNA, o2::aod::mult::MultZNC, o2::aod::mult::MultFV0M<o2::aod::mult::MultFV0A, o2::aod::mult::MultFV0C>, o2::aod::mult::MultFT0M<o2::aod::mult::MultFT0A, o2::aod::mult::MultFT0C>, o2::aod::mult::MultFDDM<o2::aod::mult::MultFDDA, o2::aod::mult::MultFDDC>, o2::aod::mult::MultTracklets, o2::aod::mult::MultTPC, o2::aod::mult::MultNTracksPV, o2::aod::mult::MultNTracksPVeta1>, o2::soa::Table<o2::aod::hf_selection_dmeson_collision::DmesonSel> > >, o2::soa::Filtered<o2::soa::Join<o2::soa::Join<o2::aod::HfCand3ProngExtExtension, o2::soa::Table<o2::soa::Index<0, -1>, o2::aod::hf_cand::CollisionId, o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex, o2::aod::hf_cand::ErrorDecayLength, o2::aod::hf_cand::ErrorDecayLengthXY, o2::aod::hf_cand::Chi2PCA, o2::aod::hf_cand::RSecondaryVertex<o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex>, o2::aod::hf_cand::DecayLength<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex>, o2::aod::hf_cand::DecayLengthXY<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex>, o2::aod::hf_cand::DecayLengthNormalised<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex, o2::aod::hf_cand::ErrorDecayLength>, o2::aod::hf_cand::DecayLengthXYNormalised<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ErrorDecayLengthXY>, o2::aod::hf_cand::ImpactParameterNormalised0<o2::aod::hf_cand::ImpactParameter0, o2::aod::hf_cand::ErrorImpactParameter0>, o2::aod::hf_cand::PtProng0<o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0>, o2::aod::hf_cand::Pt2Prong0<o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0>, o2::aod::hf_cand::PVectorProng0<o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0, o2::aod::hf_cand::PzProng0>, o2::aod::hf_cand::ImpactParameterNormalised1<o2::aod::hf_cand::ImpactParameter1, o2::aod::hf_cand::ErrorImpactParameter1>, o2::aod::hf_cand::PtProng1<o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1>, o2::aod::hf_cand::Pt2Prong1<o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1>, o2::aod::hf_cand::PVectorProng1<o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1, o2::aod::hf_cand::PzProng1>, o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0, o2::aod::hf_cand::PzProng0, o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1, o2::aod::hf_cand::PzProng1, o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2, o2::aod::hf_cand::PzProng2, o2::aod::hf_cand::ImpactParameter0, o2::aod::hf_cand::ImpactParameter1, o2::aod::hf_cand::ImpactParameter2, o2::aod::hf_cand::ErrorImpactParameter0, o2::aod::hf_cand::ErrorImpactParameter1, o2::aod::hf_cand::ErrorImpactParameter2, o2::aod::hf_track_index::Prong0Id, o2::aod::hf_track_index::Prong1Id, o2::aod::hf_track_index::Prong2Id, o2::aod::hf_track_index::HFflag, o2::aod::hf_cand_3prong::M<o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0, o2::aod::hf_cand::PzProng0, o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1, o2::aod::hf_cand::PzProng1, o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2, o2::aod::hf_cand::PzProng2>, o2::aod::hf_cand_3prong::M2<o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0, o2::aod::hf_cand::PzProng0, o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1, o2::aod::hf_cand::PzProng1, o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2, o2::aod::hf_cand::PzProng2>, o2::aod::hf_cand_3prong::ImpactParameterProngSqSum<o2::aod::hf_cand::ImpactParameter0, o2::aod::hf_cand::ImpactParameter1, o2::aod::hf_cand::ImpactParameter2>, o2::aod::hf_cand::ImpactParameterNormalised2<o2::aod::hf_cand::ImpactParameter2, o2::aod::hf_cand::ErrorImpactParameter2>, o2::aod::hf_cand::PtProng2<o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2>, o2::aod::hf_cand::Pt2Prong2<o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2>, o2::aod::hf_cand::PVectorProng2<o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2, o2::aod::hf_cand::PzProng2>, o2::aod::hf_cand::Pt<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py>, o2::aod::hf_cand::Pt2<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py>, o2::aod::hf_cand::P<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::P2<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::PVector<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::CPA<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex, o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::CPAXY<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py>, o2::aod::hf_cand::Ct<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex, o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::ImpactParameterXY<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::collision::PosZ, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ZSecondaryVertex, o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand_3prong::MaxNormalisedDeltaIP<o2::aod::collision::PosX, o2::aod::collision::PosY, o2::aod::hf_cand::XSecondaryVertex, o2::aod::hf_cand::YSecondaryVertex, o2::aod::hf_cand::ErrorDecayLengthXY, o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand::ImpactParameter0, o2::aod::hf_cand::ErrorImpactParameter0, o2::aod::hf_cand::ImpactParameter1, o2::aod::hf_cand::ErrorImpactParameter1, o2::aod::hf_cand::ImpactParameter2, o2::aod::hf_cand::ErrorImpactParameter2, o2::aod::hf_cand::PxProng0, o2::aod::hf_cand::PyProng0, o2::aod::hf_cand::PxProng1, o2::aod::hf_cand::PyProng1, o2::aod::hf_cand::PxProng2, o2::aod::hf_cand::PyProng2>, o2::aod::hf_cand::Eta<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::Phi<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py>, o2::aod::hf_cand::Y<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::E<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz>, o2::aod::hf_cand::E2<o2::aod::hf_cand_3prong::Px, o2::aod::hf_cand_3prong::Py, o2::aod::hf_cand_3prong::Pz> > >, o2::soa::Table<o2::aod::hf_sel_candidate_dplus::IsSelDplusToPiKPi>, o2::soa::Table<o2::aod::hf_cand_3prong::FlagMcMatchRec, o2::aod::hf_cand_3prong::OriginMcRec, o2::aod::hf_cand_3prong::IsCandidateSwapped, o2::aod::hf_cand_3prong::FlagMcDecayChanRec> > >, o2::soa::Filtered<o2::soa::Join<o2::soa::Join<o2::aod::TracksExtension, o2::soa::Table<o2::soa::Index<0, -1>, o2::aod::track::CollisionId, o2::aod::track::TrackType, o2::aod::track::X, o2::aod::track::Alpha, o2::aod::track::Y, o2::aod::track::Z, o2::aod::track::Snp, o2::aod::track::Tgl, o2::aod::track::Signed1Pt, o2::aod::track::IsWithinBeamPipe<o2::aod::track::X>, o2::aod::track::Px<o2::aod::track::Signed1Pt, o2::aod::track::Snp, o2::aod::track::Alpha>, o2::aod::track::Py<o2::aod::track::Signed1Pt, o2::aod::track::Snp, o2::aod::track::Alpha>, o2::aod::track::Pz<o2::aod::track::Signed1Pt, o2::aod::track::Tgl>, o2::aod::track::Energy<o2::aod::track::Signed1Pt, o2::aod::track::Tgl>, o2::aod::track::Rapidity<o2::aod::track::Signed1Pt, o2::aod::track::Tgl>, o2::aod::track::Sign<o2::aod::track::Signed1Pt>, o2::soa::Marker<1> > >, o2::soa::Table<o2::aod::track::DcaXY, o2::aod::track::DcaZ> > > >::GroupedCombinationsGenerator(<brace-enclosed initializer list>)'
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/O2Physics/2329/791bef733ef0558b52dda12f3c70c75d8f56cfcb/build_O2Physics_o2/pretty.html).

vkucera commented:
Hi @Simple-Shyam , I don't see any update since my review and the code does not compile. What is the status?

Simple-Shyam commented:
> Hi @Simple-Shyam , I don't see any update since my review and the code does not compile. What is the status?

The code compiled initially, but then there is a slice cache method pushed on the repository, and then it doesn't compile because of this new method. I implemented the comments and slice cache method, and then there was an issue with findBin (returning an unknown number), which I reported on Mattermost. I updated O2Physics, and it fixes the findBin issue. I am currently checking the results' compatibility and will commit it again in a few days.

vkucera commented:
> > Hi @Simple-Shyam , I don't see any update since my review and the code does not compile. What is the status?
> 
> The code compiled initially, but then there is a slice cache method pushed on the repository, and then it doesn't compile because of this new method. I implemented the comments and slice cache method, and then there was an issue with findBin (returning an unknown number), which I reported on Mattermost. I updated O2Physics, and it fixes the findBin issue. I am currently checking the results' compatibility and will commit it again in a few days.

OK, no problem. Next time when you expect to take several days to update the PR, please convert it into a draft until you push new commits.

vkucera commented:
Hi @Simple-Shyam , can you please explain why you closed this PR and opened a new identical one? https://github.com/AliceO2Group/O2Physics/pull/2561

Simple-Shyam commented:
> Hi @Simple-Shyam , can you please explain why you closed this PR and opened a new identical one? #2561

Thanks Vit, I implemented the comments, you mentioned above. I also added the event mixing for the MCGen mode. I didn't close it. I just updated my local repository before committing the new locally compiled codes with recent O2Physics, and then the pull request is automatically de-attached from my local repository and marked as close. I don't know what is the proper solution in that case, so I created a fresh pull request. You can suggest me the best solution in this case.

vkucera commented on PWGHF/DataModel/CandidateReconstructionTables.h:
```diff
@@ -1290,15 +1290,24 @@ DECLARE_SOA_COLUMN(PtD, ptD, float);
 DECLARE_SOA_COLUMN(PtHadron, ptHadron, float);
 DECLARE_SOA_COLUMN(MD, mD, float);
 DECLARE_SOA_COLUMN(SignalStatus, signalStatus, int);
+DECLARE_SOA_COLUMN(PoolBin, poolBin, int);
```
Still missing for `PoolBin`.

Simple-Shyam replied:
added also for poolBin.

vkucera commented:
> > Hi @Simple-Shyam , can you please explain why you closed this PR and opened a new identical one? #2561
> 
> Thanks Vit, I implemented the comments, you mentioned above. I also added the event mixing for the MCGen mode. I didn't close it. I just updated my local repository before committing the new locally compiled codes with recent O2Physics, and then the pull request is automatically de-attached from my local repository and marked as close. I don't know what is the proper solution in that case, so I created a fresh pull request. You can suggest me the best solution in this case.

Hi @Simple-Shyam , several comments are still not addressed.

Simple-Shyam commented on PWGHF/DataModel/CandidateReconstructionTables.h:
```diff
@@ -1290,15 +1290,24 @@ DECLARE_SOA_COLUMN(PtD, ptD, float);
 DECLARE_SOA_TABLE(DplusHadronRecoInfo, "AOD", "DPLUSHRECOINFO",
                   aod::hf_correlation_dplus_hadron::MD,
                   aod::hf_correlation_dplus_hadron::SignalStatus);
+// Table for selection of Dmeson in a collision
```
Yes done.

fcolamar commented:
Ciao @Simple-Shyam, I see that you pushed further updates, but without commenting: do you confirm that you took into consideration the remaining comments by @vkucera, or is something still remaining? Can we in the former case proceed forward?
Cheers,
Fabio

Simple-Shyam commented:
> Ciao @Simple-Shyam, I see that you pushed further updates, but without commenting: do you confirm that you took into consideration the remaining comments by @vkucera, or is something still remaining? Can we in the former case proceed forward? Cheers, Fabio

Ciao @fcolamar  yes, I updated the code implementing the comments and informed @vkucera in matter-most in the personal message. I was waiting for reply. In the mean time base CandidateReconstructionTables.h class is changed. I have to start with the new CandidateReconstructionTables.h again. May be I will do it later.

Thank you
Shyam

Simple-Shyam commented on PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx:
```diff
@@ -19,12 +19,16 @@
 #include "Framework/runDataProcessing.h"
+#include "Common/DataModel/EventSelection.h"
+#include "Common/DataModel/Centrality.h"
+#include "Common/DataModel/Multiplicity.h"
```
I sorted all the includes.

vkucera replied:
You should sort the paths and keep PWGHF last. See the HF guidelines.

Simple-Shyam replied:
For me, this should be sort version. I need to sort w.r.t. Path. Is it fine?
#include "Common/Core/TrackSelection.h"
#include "Common/DataModel/TrackSelectionTables.h"
#include "Framework/AnalysisTask.h"
#include "Framework/HistogramRegistry.h"
#include "Framework/runDataProcessing.h"
#include "PWGHF/DataModel/CandidateReconstructionTables.h"
#include "PWGHF/DataModel/CandidateSelectionTables.h"

vkucera replied:
Yes this would be fine

Simple-Shyam replied:
Thanks.

Simple-Shyam commented on PWGHF/DataModel/CandidateReconstructionTables.h:
```diff
@@ -1285,22 +1285,32 @@ DECLARE_SOA_TABLE(DsHadronRecoInfo, "AOD", "DSHRECOINFO", //! Ds-Hadrons pairs R
+DECLARE_SOA_COLUMN(PtD, ptD, float);                 //! Transverse momentum of D+
+DECLARE_SOA_COLUMN(PtHadron, ptHadron, float);       //! Transverse momentum of Hadron
+DECLARE_SOA_COLUMN(MD, mD, float);                   //! Invariant mass of D+
+DECLARE_SOA_COLUMN(SignalStatus, signalStatus, int); //! Used in MC-Rec, D+ Signal
```
Yes done

vkucera commented:
Hi @Simple-Shyam , there are still pending comments, that have not been addressed.

Simple-Shyam commented:
> Hi @Simple-Shyam , there are still pending comments, that have not been addressed.

Thanks, Can you please highlight the comment/comments? Next time I would try to do shorter commit. Is there any way, I can check which comment is not addressed. I am a bit confused why comments are missing? I read two times.

vkucera commented:
> > Hi @Simple-Shyam , there are still pending comments, that have not been addressed.
> 
> Thanks, Can you please highlight the comment/comments? Next time I would try to do shorter commit. Is there any way, I can check which comment is not addressed. I am a bit confused why comments are missing from me? I read two times.

You see that the comments on this page marked as resolved are collapsed. The unresolved comments are still fully visible.

Simple-Shyam commented:
> s on this page marked as resolved are collapsed

Thanks now it's clear. I need to check the flag that can tell me the situation. I am not very much expert so taking time.

vkucera reviewed (approved):
Thanks @Simple-Shyam , it looks fine to me.
@mmazzilli @deepathoms please have a look.

Simple-Shyam commented:
> Thanks @Simple-Shyam , it looks fine to me. @mmazzilli @deepathoms please have a look.

Thank you @vkucera for helping to learn the things.

deepathoms commented:
The commit looks good for me. Green light to merge it.
